backend/common/src/main/java/ai/verta/modeldb/common/configuration/GracefulShutdownBeans.java

```diff
@@ -0,0 +1,27 @@
+package ai.verta.modeldb.common.configuration;
+
+import ai.verta.modeldb.common.GracefulShutdown;
+import ai.verta.modeldb.common.config.Config;
+import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory;
+import org.springframework.boot.web.servlet.server.ServletWebServerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class GracefulShutdownBeans {
+
+  @Bean
+  public GracefulShutdown gracefulShutdown(Config config) {
+    if (config == null || config.getSpringServer().getShutdownTimeout() == null) {
```

Comment:
I don't think Spring will let this `config` be null, will it?

Reply:
Resolved